soc/arc/snps_arc_hsdk4xd/CMakeLists.txt

```
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(COMPILER STREQUAL gcc)
+  # GNU compiler options
+  zephyr_compile_options(-mno-sdata)
+  zephyr_compile_options_ifdef(CONFIG_FPU -mfpu=fpud_all)
+else()
+  # MWDT compiler options
+   zephyr_compile_options(-arcv2hs -core4 -Xdual_issue -Xcode_density -Hrgf_banked_regs=32 -HL
+    -Xatomic -Xll64 -Xunaligned -Xdiv_rem=radix4 -Xswap -Xbitscan -Xmpy_option=qmpyh -Xshift_assist
+    -Xbarrel_shifter -Xtimer0 -Xtimer1 -Xrtc -Hld_cycles=2 -Xlpb_size=128)
```

```suggestion
  zephyr_compile_options(-arcv2hs -core4 -Xdual_issue -Xcode_density -Hrgf_banked_regs=32 -HL
    -Xatomic -Xll64 -Xunaligned -Xdiv_rem=radix4 -Xswap -Xbitscan -Xmpy_option=qmpyh -Xshift_assist
    -Xbarrel_shifter -Xtimer0 -Xtimer1 -Xrtc -Hld_cycles=2)
```

Alignment fix and drop Xlpb_size as we disable LPB anyway.